core/echo_errors.go

```diff
@@ -0,0 +1,121 @@
+package core
+
+import (
+	"errors"
+	"fmt"
+	"github.com/labstack/echo/v4"
+	"github.com/nuts-foundation/nuts-node/didman/logging"
+	"net/http"
+	"schneider.vip/problem"
+)
+
+const statusCodesContextKey = "!!ErrorStatusCodes"
+const operationIdContextKey = "!!OperationId"
+
+func createHTTPErrorHandler(routers []Routable) echo.HTTPErrorHandler {
+	globalErrorStatusCodes := errorStatusCodesFromRouters(routers)
+	return func(err error, ctx echo.Context) {
+		if echoErr, ok := err.(*echo.HTTPError); ok {
+			ctx.Echo().DefaultHTTPErrorHandler(echoErr, ctx)
+			return
+		}
+		title := fmt.Sprintf("%s failed", fmt.Sprintf("%s", ctx.Get(operationIdContextKey)))
+		statusCode := getHTTPStatusCode(err, globalErrorStatusCodes, errorStatusCodesFromContext(ctx))
+		result := problem.New(problem.Title(title), problem.Status(statusCode), problem.Detail(err.Error()))
+		if statusCode == http.StatusInternalServerError {
+			logging.Log().WithError(err).Error(title)
+		} else {
+			logging.Log().WithError(err).Warn(title)
+		}
+		if !ctx.Response().Committed {
+			if _, err := result.WriteTo(ctx.Response()); err != nil {
+				ctx.Echo().Logger.Error(err)
+			}
+		} else {
+			ctx.Echo().Logger.Warnf("Unable to send error back to client, response already committed: %v", err)
+		}
+	}
+}
+
+func NotFoundError(errStr string, args ...interface{}) HTTPStatusCodeError {
```

Comment:
you could have these funcs return the `error` interface and make `HTTPStatusCodeError` private. That would make it less confusing for others to use it. (or misuse it)